contrib/maintenance-es-index.sh

```diff
@@ -16,27 +17,37 @@ index=metrictank
 
 # show all matching documents, given a query
 function es_search () {
-	curl "http://localhost:9200/$index/metric_index/_search?q=$1&pretty=true"
+    size=${2:-20}
+    curl -s "http://localhost:9200/$index/metric_index/_search?q=$1&pretty=true&size=$size"
 }
 
 # delete all matching documents, given a query
 function es_delete () {
-        curl -X DELETE "http://localhost:9200/$index/metric_index/_query?q=$1"
+    curl -s -X DELETE "http://localhost:9200/$index/metric_index/_query?q=$1"
+}
+
+# list series matching a graphite pattern
+function es_list () {
+    es_search $(es_query $1) 10000 | grep name | sed -e 's/^.*: "\|",$//g'
 }
 
 # construct a query, given a graphite pattern
 # e.g. so*.*.and.so -> nodes.n0:so*+AND+nodes.n2:and+AND+nodes.n3:so
 # caveats:
-# - foo.*.* will also match foo or foo.* or foo.*.*.* since we don't have a check yet for number of fields
 # - does not support {foo,bar} or [this,kindof] graphite syntax! just literal words and wildcards.
 function es_query () {
     local q
     IFS='.' read -ra nodes <<< "$1"
     for i in "${!nodes[@]}"; do
-        [ "${nodes[$i]}" == '*' ] && continue
-        new="nodes.n$i:${nodes[$i]}"
-        [ -n "$q" ] && q="$q+AND+"
+        if [ "${nodes[$i]}" == '*' ]; then
+            new="%2B_exists_:nodes.n$i"
+        else
+            new="%2Bnodes.n$i:%22${nodes[$i]}%22"
+        fi
+        [ -n "$q" ] && q="$q+"
         q="$q$new"
     done
+    i=$(expr $i + 1)
+    q="$q+-_exists_:nodes.n$i"
```

Comment:
Do i understand correctly that this checks for the *absence* of the "next word we don't need" ? e.g. for `foo.*` it checks for absence of 3rd node? that's neat.